File: ms-auth/src/auth/domain/model/value-objects/email.ts
const EMAIL_REGEX = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;

export class Email {
  private constructor(public readonly value: string) {}

  static of(raw: string): Email {
    if (!raw || typeof raw !== 'string') {
      throw new Error('Email cannot be empty');
    }
    const normalized = raw.trim().toLowerCase();
    if (!EMAIL_REGEX.test(normalized)) {
      throw new Error(`Invalid email format: ${raw}`);
    }
    return new Email(normalized);
  }

  equals(other: Email): boolean {
    return this.value === other.value;
  }

  toString(): string {
    return this.value;
  }
}

File: ms-doctor/src/main/java/com/encuentratumedico/msdoctor/domain/model/shared/Email.java
package com.encuentratumedico.msdoctor.domain.model.shared;

import java.util.regex.Pattern;

public record Email(String value) {
    private static final Pattern EMAIL_REGEX = Pattern.compile("^[^\\s@]+@[^\\s@]+\\.[^\\s@]+$");

    public Email {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("Email cannot be empty");
        }
        value = value.trim().toLowerCase();
        if (!EMAIL_REGEX.matcher(value).matches()) {
            throw new IllegalArgumentException("Invalid email format: " + value);
        }
    }

    public static Email of(String raw) {
        return new Email(raw);
    }

    public static Email ofNullable(String raw) {
        if (raw == null || raw.isBlank()) return null;
        return new Email(raw);
    }
}
